lib/cisco_node_utils/syslog_server.rb

```diff
@@ -42,12 +52,20 @@ def initialize(name,
 
     def self.syslogservers
       hash = {}
-
       syslogservers_list = config_get('syslog_server', 'server')
       return hash if syslogservers_list.nil?
 
       syslogservers_list.each do |id|
+        # The YAML regex isn't specific enough for some platforms,
+        # so we have to do further checking.
+        begin
+          IPAddr.new(id)
+        rescue
+          next
+        end
+
         level = config_get('syslog_server', 'level', id)
+        level = LEVEL_TO_NUM[level] if platform == :ios_xr
 
         vrf = config_get('syslog_server', 'vrf', id)
```

Comment:
Looks like this method needs to be redesigned for XR:

```
hash[id] = SyslogServer.new(id, level, vrf, false)
```

You can have the same server (id) configured in multiple VRFs in XR:

```
logging 1.2.3.4 vrf red severity info port default
logging 1.2.3.4 vrf default severity info port default
```

but you're trying to store both instances under the same id (1.2.3.4) in this method.